pkg/cloudclient/aws/private.go

```diff
@@ -62,35 +64,85 @@ var (
 	userdataEndVerifier   string = "USERDATA END"
 )
 
-func newClient(ctx context.Context, logger ocmlog.Logger, accessID, accessSecret, sessiontoken, region, instanceType string, tags map[string]string) (*Client, error) {
-	cfg, err := config.LoadDefaultConfig(ctx,
-		config.WithRegion(region),
-		config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
-			Value: aws.Credentials{
-				AccessKeyID: accessID, SecretAccessKey: accessSecret, SessionToken: sessiontoken,
-			},
-		}),
-	)
+func getEc2ClientFromInput(input ClientInput) (ec2.Client, error) {
+	var cfg aws.Config
+	var err error
+	var ec2Client ec2.Client
+	//Get config from profile if provided
+	if input.Profile != "" {
+		cfg, err = config.LoadDefaultConfig(input.Ctx,
+			config.WithSharedConfigProfile(input.Profile),
+			config.WithRegion(input.Region),
+		)
+		if err != nil {
+			return ec2Client, fmt.Errorf("can not get AWS config from profile `%s`", input.Profile)
+		}
+
+	} else {
+		//Get config from access keys if provided
+		cred := credentials.NewStaticCredentialsProvider(input.AccessKeyId,
```

Comment:
```go
		//Get config from access keys if provided
		cred := credentials.NewStaticCredentialsProvider(input.AccessKeyId,
			input.SecretAccessKey, input.SessionToken)
		switch cr := interface{}(cred).(type) {
```
are you sure the above is looking right? you're getting `credentials.NewStaticCredentialsProvider` from "github.com/aws/aws-sdk-go-v2/credentials", so your switch-case would always catch `case awscredsv2.StaticCredentialsProvider`

Reply:
in `cmd/`, we fixed it into v2.
But in `pkg/`, NewClient() supports both v1/v2: https://github.com/openshift/osd-network-verifier/blob/main/pkg/cloudclient/aws/aws.go#L54 (as `creds interface{}`)

Reply:
We can remove v1 completely, but I've didn't want to introduce that change in this PR. Can we card it out?

Reply:
@boranx on a second look, I see what you were suggesting. I removed the switch loop. I don't see a need for it any more. Downstream can pass in creds and onv can create ec2client object regardless of version.

Reply:
ack, that looks cleaner :) I think eventually we'd need a jira card to make sure this is imported to uhc-backend gracefully
(As they no more need creds, instead it's just: `cli, err := cloudclient.NewClient(context.TODO(), logger, cmdOptions)`)

Reply:
I'm adding it it as a todo in this card before promoting to prod